cekit/cli.py

```diff
@@ -171,31 +171,36 @@ def run(self):
             logger.warning("You are running unreleased development version of Cekit, "
                            "use it only at your own risk!")
 
-        try:
-            tools.cfg = tools.get_cfg(self.args.config)
-            tools.cleanup(self.args.target)
-
-            if self.args.redhat:
-                tools.cfg['common']['redhat'] = True
-            if self.args.work_dir:
-                tools.cfg['common']['work_dir'] = self.args.work_dir
-
-            # We need to construct Generator first, because we need overrides
-            # merged in
-            params = {'redhat': tools.cfg['common']['redhat']}
-            generator = Generator(self.args.descriptor,
-                                  self.args.target,
-                                  self.args.build_engine,
-                                  self.args.overrides,
-                                  params)
+        tools.cfg = tools.get_cfg(self.args.config)
+        tools.cleanup(self.args.target)
+
+        if self.args.redhat:
+            tools.cfg['common']['redhat'] = True
+        if self.args.work_dir:
+            tools.cfg['common']['work_dir'] = self.args.work_dir
 
-            # Now we can fetch repositories of modules (we have all overrides)
-            get_dependencies(generator.image,
-                             os.path.join(self.args.target, 'repo'))
+        # We need to construct Generator first, because we need overrides
+        # merged in
+        params = {'redhat': tools.cfg['common']['redhat']}
+        self.generator = Generator(self.args.descriptor,
+                                   self.args.target,
+                                   self.args.build_engine,
+                                   self.args.overrides,
+                                   params)
 
-            # We have all overrided repo fetch so we can discover modules
-            # and process its dependency trees
-            discover_modules(os.path.join(self.args.target, 'repo'))
+        # Now we can fetch repositories of modules (we have all overrides)
+        get_dependencies(self.generator.image,
```

Comment:
this is  cloning the module repos I would postpone that to run method.

Reply:
Thanks, good point: force-pushed a rebase to fix.